### PR TITLE
Body parsers

### PR DESCRIPTION
**Pull request by jamesgorrie**
We now have an array of body parts.
The order is:
* Get the body
* Turn it into a fragment
* recursively filter / clean the nodes and their children
* Split it into a 1 dimensional array

The stuff coming from the API is really simple - and seems to only allow `p`s and `h2`s - so we shouldn't need much more for now - but once things like `blockquote`s (#80) get in we'll have to start not just filtering but parsing.

### CODE REVIEW

**Comment by jamesgorrie** [comment]
Ah yes - some testing has been added too.

File: server/util/body-parser.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+// This recursively cleans the nodes.
+function cleanNodes(nodes) {
+  return removeEmptyTextNodes(nodes).map(node => {
+    if (node.childNodes && node.childNodes.length > 0) {
+      const childNodes = cleanNodes(node.childNodes);
+      return Object.assign({}, node, {childNodes});
+    } else {
+        return node;
```
Comment (davidpmccormick) [inline server/util/body-parser.js]: This indent makes me feel uneasy!

**Comment by gestchild** [comment]
Is it worth removing classes from nodes at this point?

**Comment by jamesgorrie** [comment]
@gestchild 
> Is it worth removing classes from nodes at this point?

I haven't seen any that have any classes - but if we did, yes this is where it would be.
We can have that as another CR.

**Pull request merged**